fe/fe-core/src/main/java/com/starrocks/connector/hive/CacheUpdateProcessor.java

```diff
@@ -85,17 +90,70 @@ public void refreshTable(String dbName, Table table, boolean onlyCachedPartition
+    public void refreshTableBackground(Table table, boolean onlyCachedPartitions, ExecutorService executor) {
         HiveMetaStoreTable hmsTbl = (HiveMetaStoreTable) table;
-        metastore.refreshTable(hmsTbl.getDbName(), hmsTbl.getTableName(), onlyCachedPartitions);
-        refreshRemoteFiles(hmsTbl.getTableLocation(), Operator.UPDATE, getExistPaths(hmsTbl), onlyCachedPartitions,
-                executor);
+        if (metastore.refreshTableBackground(hmsTbl.getDbName(), hmsTbl.getTableName(), onlyCachedPartitions)) {
+            Map<BasePartitionInfo, Partition> updatedPartitions = getUpdatedPartitions(hmsTbl);
+            if (!updatedPartitions.isEmpty()) {
+                // update partition remote files cache
+                List<String> updatedPaths = updatedPartitions.values().stream().map(Partition::getFullPath)
+                        .map(path -> path.endsWith("/") ? path : path + "/")
+                        .collect(Collectors.toList());
+                refreshRemoteFilesBackground(hmsTbl.getTableLocation(), updatedPaths, onlyCachedPartitions, executor);
+
+                LOG.info("{}.{}.{} partitions has updated, updated partition size is {}, " +
+                                "refresh partition and file success", hmsTbl.getCatalogName(), hmsTbl.getDbName(),
+                        hmsTbl.getTableName(), updatedPartitions.size());
+            }
+
+            // update partitionUpdatedTimes
+            updatedPartitions.entrySet().stream().filter(entry -> entry.getValue().getModifiedTime() != 0).
+                    forEach(entry -> partitionUpdatedTimes.put(entry.getKey(), entry.getValue().getModifiedTime()));
+            Map<HivePartitionName, Partition> cachedPartitions = metastore.getAllCachedPartitions();
+            partitionUpdatedTimes.keySet().removeIf(basePartitionInfo -> !cachedPartitions.containsKey(
+                    HivePartitionName.of(basePartitionInfo.dbName, basePartitionInfo.tableName,
+                            basePartitionInfo.partitionName)));
+        }
     }
 
     public Set<HiveTableName> getCachedTableNames() {
         return ((CachingHiveMetastore) metastore).getCachedTableNames();
     }
 
+    private Map<BasePartitionInfo, Partition> getUpdatedPartitions(HiveMetaStoreTable table) {
+        String dbName = table.getDbName();
+        String tblName = table.getTableName();
+
+        Map<BasePartitionInfo, Partition> toCheckUpdatedPartitionInfoMap = Maps.newHashMap();
+        if (table.isUnPartitioned()) {
+            Partition partition = metastore.getPartition(dbName, tblName, Lists.newArrayList());
+            BasePartitionInfo partitionInfo = new BasePartitionInfo(dbName, tblName, tblName);
+            toCheckUpdatedPartitionInfoMap.put(partitionInfo, partition);
+        } else {
+            Map<HivePartitionName, Partition> partitions = metastore.getCachedPartitions(dbName, tblName);
```

Comment:
This is already got by `refreshTable`. Try to reuse the result instead of getting it twice.
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/8065637/229581099-c13d604c-93ce-4a13-ad04-9bc8ef751ab7.png">